### DISCUSSION TITLE
Quicksort and Traits examples from the webpage don't compile

### DISCUSSION THREAD

**Discussion by sparky4pro** [julelang/jule#22] [Discussions]
Hi, started to look into Jule and wanted to let you know that the quicksort example on the webpage does not compile.
It differs from the quicksort example from the test directory, which compiles just fine.
Suppose the webpage code needs to be updated.

Also, the traits example does not compile either.
There is no error message, but the compiler says: "exit status 1".
(webpage version and test directory version are the same)

Besides that, keep up the good work!
sparky

**Comment by mertcandav** [comment]
Hi, thank you for your interest.

Compiler rewritten, here is PR: #23
Do the same issues persist with the new compiler?

**Comment by sparky4pro** [comment]
Hello, downloaded the new version, tested a few files, and they compile fine.
Even the 'traits' example :D

Didn't fully understand the traits example, probably because I don't really understand the difference between interfaces and traits yet.
Have to re-read the Jule documentation again...

What I see so far is really interesting.
I might post more questions in the future, and probably annoy you with even more questions...

Thank you.

**Reply by mertcandav** [reply]
Jule's traits are currently a characterized mask of the methods a structure should implement. When used as a trait data type, all structures that have applied that trait are considered compatible. In many ways, this can resemble the "interface" approaches in different languages. This can be called a nomenclature for now. Jule's traits currently do not have a very distinctive difference.

Thanks for your attention, every question is appreciated.
